src/WebJobs.Script/Workers/Rpc/Configuration/RpcWorkerConfigFactory.cs

```diff
@@ -118,6 +119,24 @@ internal void AddProvider(string workerDir)
                     RpcWorkerDescription workerDescription = workerConfig.Property(WorkerConstants.WorkerDescription).Value.ToObject<RpcWorkerDescription>();
                     workerDescription.WorkerDirectory = workerDir;
 
+                    //Overwrite default Description with profile
+                    descriptionProfiles = GetWorkerDescriptionProfiles(workerConfig);
+
+                    if (_environment.IsConsumptionSku())
+                    {
+                        if (!_environment.IsAppInsightsAgentOptOut())
+                        {
+                            workerDescription = GetWorkerDescriptionFromProfiles(RpcWorkerConstants.WorkerDescriptionAppInsightsAgentEnvironmentProfileName, descriptionProfiles, workerDescription);
+                        }
+                    }
+                    else
+                    {
+                        if (_environment.IsAppInsightsAgentEnabled())
+                        {
```

Comment:
@shreyas-gopalakrishna - We should not be relying on the App Setting  here. @liliankasem - can you please share notes from the meeting on approach discussed?